packages/blade/src/components/Button/BaseButton/BaseButton.stories.tsx

```diff
@@ -0,0 +1,68 @@
+import type { ComponentStory, Meta } from '@storybook/react';
+import {
+  Title,
+  Subtitle,
+  Primary,
+  ArgsTable,
+  Stories,
+  PRIMARY_STORY,
+  Description,
+} from '@storybook/addon-docs';
+import iconMap from '../../Icons/iconMap';
+import type { BaseButtonIcon, BaseButtonProps } from './BaseButton';
+import BaseButtonComponent from './BaseButton';
+
+export default {
+  title: 'Components/Button/BaseButton (Internal)',
+  component: BaseButtonComponent,
+  args: {
+    children: 'Pay Now',
+  },
+  argTypes: {
+    icon: {
+      name: 'icon',
+      type: 'select',
+      options: Object.keys(iconMap),
+    },
+  },
+  parameters: {
+    docs: {
+      page: () => (
+        <>
+          <Title />
+          <Subtitle>
+            This is the BaseButton component. It is only for internal Blade usage.
+          </Subtitle>
+          <Title>Example</Title>
+          <Primary />
+          <Title>Properties</Title>
+          <Description
+            markdown="
+             >For demonstration purposes, we have created a dropdown of icons to choose from in the `icon` prop. The `icon` prop expects the actual Icon component of type `IconComponentType` to be passed. 
+             "
+          />
+          <ArgsTable story={PRIMARY_STORY} />
+          <Stories />
+        </>
+      ),
+    },
+  },
+} as Meta<BaseButtonProps<{ icon: BaseButtonIcon }>>;
+
+const BaseButtonTemplate: ComponentStory<typeof BaseButtonComponent> = ({
+  icon,
+  children,
+  ...args
+}) => {
+  const IconComponent = iconMap[(icon as unknown) as string];
```

Comment:
@chaitanyadeorukhkar ??

Reply:
Not sure what you mean here. I'm changing the type here since the actual `icon` prop is of type `IconComponent` but here in the stories I'm using it as a `string` to give a better experience for when developers are trying icons in the story